test/PerformanceTests/SerializationComparisonsTests/JsonWriterBenchmarks/Benchmarks.cs

```diff
@@ -68,23 +65,29 @@ public async Task WriteToMemoryAsync()
         public async Task WriteToFileAsync()
         {
             // multiple writes to increase benchmark duration
-            await writer.WritePayloadAsync(data, fileStream);
-            await writer.WritePayloadAsync(data, fileStream);
-            await writer.WritePayloadAsync(data, fileStream);
-            await writer.WritePayloadAsync(data, fileStream);
-            await writer.WritePayloadAsync(data, fileStream);
+            await WritePayloadAsync();
+            await WritePayloadAsync();
+            await WritePayloadAsync();
+            await WritePayloadAsync();
+            await WritePayloadAsync();
         }
 
         [Benchmark]
         [BenchmarkCategory("RawValues")]
         public async Task WriteWithRawValues()
         {
             // multiple writes to increase benchmark duration
-            await writer.WritePayloadAsync(data, fileStream, includeRawValues: true);
-            await writer.WritePayloadAsync(data, fileStream, includeRawValues: true);
-            await writer.WritePayloadAsync(data, fileStream, includeRawValues: true);
-            await writer.WritePayloadAsync(data, fileStream, includeRawValues: true);
-            await writer.WritePayloadAsync(data, fileStream, includeRawValues: true);
+            await WritePayloadAsync(includeRawValues: true);
+            await WritePayloadAsync(includeRawValues: true);
+            await WritePayloadAsync(includeRawValues: true);
+            await WritePayloadAsync(includeRawValues: true);
+            await WritePayloadAsync(includeRawValues: true);
+        }
+
+        private async Task WritePayloadAsync(bool includeRawValues = false)
+        {
+            using var outputStream = new FileStream(filePath, FileMode.Create, FileAccess.Write, share: FileShare.ReadWrite);

```

Comment:
Initializing the file stream from here won't skew the results?

Reply:
It may skew the results, but since it's execute for every writer, I think it still makes for a fair comparison. And the relative runtimes and allocations of the writers in the results did not change (as compared to each other), so I think the results are not less reliable than they used to be. But now the tests are cleaner (the previous version of the tests did not delete the temporary file)